### PR TITLE
Add message handler for Clair messages

### PR DESCRIPTION
**Pull request by Allda**
Clair is a new service for container grading. This commit adds handlers
for all Clair + wrapper messages.

### CODE REVIEW

File: fedmsg_meta_umb/clair.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            if inner_msg.get('brew_build'):
+                return 'https://brewweb.engineering.redhat.com/brew/search?' \
+                    'terms={brew_build}&type=build&match=glob'.format(**inner_msg)
+
+        return None
+
+    def packages(self, msg, **config):
+        inner_msg = msg['msg']
+        if msg['topic'].endswith('clair.notification'):
+            return set([inner_msg['Notification']['Name']])
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: I don't think the notification uuid really qualifies as a "package".

File: fedmsg_meta_umb/clair.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        elif msg['topic'].endswith('clair.scan'):
+            return set(['{}.{}'.format(inner_msg['nvr'], inner_msg['arch'])])
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: The package is just the N of the NVR. To get it, you can do `inner_msg['nvr'].rsplit('-', 2)[0]`. You don't need to include the arch at all.

File: fedmsg_meta_umb/clair.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        elif msg['topic'].endswith('clair.scan.isv'):
+            return set(['{}'.format(inner_msg['image_id'])])
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: The `image_id` isn't really useful as a package either. Probably just want to return `set()` here.

File: fedmsg_meta_umb/clair.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        elif msg['topic'].endswith('clair.command'):
+            if inner_msg.get('brew_build'):
+                return {'{}.{}'.format(
+                    inner_msg['brew_build'], inner_msg['architecture'])}
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: `brew_build` values are also NVRs. See the comment above about extracting the N from the NVR.

File: fedmsg_meta_umb/clair.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            elif inner_msg.get('imageDigest'):
+                return {inner_msg['imageDigest']}
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: Also not a package. Just return `set()`.

File: fedmsg_meta_umb/clair.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        else:
+            return set([])
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: Don't really need the `[]` here.

File: fedmsg_meta_umb/clair.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    def objects(self, msg, **config):
```
Comment (mikebonnet) [inline fedmsg_meta_umb/clair.py]: `objects()` isn't really useful, and could be removed.

**Comment by Allda** [comment]
@mikebonnet I put those notification IDs and image ID under the package to be able to filter messages in datagrepper using queries. If you think it should not be under `package` is there any way how to extract "primary key" of the message and be alble filter it using datagrepper?

**Comment by mikebonnet** [comment]
> @mikebonnet I put those notification IDs and image ID under the package to be able to filter messages in datagrepper using queries. If you think it should not be under `package` is there any way how to extract "primary key" of the message and be alble filter it using datagrepper?

There isn't a great way to do that right now. The best option would probably be to use the `topic` and `contains` filters, though you'd want to do some additional validation because there is the possibility for false-positives. We can look into some better filtering options in the future (maybe using the `objects` meta field) but I'd rather not abuse the `package` field this way.

File: fedmsg_meta_umb/clair.py (hunk 2 above)
Comment (Allda) [reply fedmsg_meta_umb/clair.py]: fixed

File: fedmsg_meta_umb/clair.py (hunk 3 above)
Comment (Allda) [reply fedmsg_meta_umb/clair.py]: fixed

File: fedmsg_meta_umb/clair.py (hunk 4 above)
Comment (Allda) [reply fedmsg_meta_umb/clair.py]: fixed

File: fedmsg_meta_umb/clair.py (hunk 5 above)
Comment (Allda) [reply fedmsg_meta_umb/clair.py]: fixed

File: fedmsg_meta_umb/clair.py (hunk 6 above)
Comment (Allda) [reply fedmsg_meta_umb/clair.py]: removed

**Comment by mikebonnet** [comment]
This looks great, thanks for the updates!

**Pull request merged**